sockeye/inference.py

```diff
@@ -1823,17 +1836,19 @@ def _beam_search(self,
                 best_hyp_indices, best_word_indices, scores_accumulated = self._top(scores, target_dists, finished)
             else:
                 # On the first timestep, all hypotheses have identical histories, so force topk() to choose extensions
-                # of the first row only
+                # of the first row only by setting all other rows to inf
                 if t == 1 and not self.skip_topk:
-                    scores = scores[self.batch_indices, :]
```

Comment:
this change is actually creating a little bit of a performance hit. It is needed to abstract batch size away from the TopK HybridBlock. Does anyone have a better idea on how to do this?

Reply:
using a mask, allocated before the beam search iteration is faster. Still a tiny slowdown for larger batch sizes (-0.1 sent/sec), but that should be fine.